agent/api/task.go

```diff
@@ -839,6 +859,23 @@ func (task *Task) GetTaskENI() *ENI {
 	return task.ENI
 }
 
+// ShouldWaitForExecutionCredentials check if there are container waiting for the
+// credentials to progress eg: pull
+func (task *Task) ShouldWaitForExecutionCredentials() bool {
+	if task.GetDesiredStatus() == TaskStopped || task.GetKnownStatus() > TaskStatusNone {
```

Comment:
Can you add some documentation for why/what the `task.GetKnownStatus() > TaskStatusNone` check is doing?